File: Zipping and Unzipping files in .NET application/[C#]-Zipping and Unzipping files in .NET application/C#/Password.cs
using System;
using System.Windows.Forms;
using ComponentPro.Compression;

namespace ArchiveManager
{
    public partial class Password : Form
    {
        public Password()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Gets or sets the archive's password.
        /// </summary>
        public string Pass
        {
            get
            {
                return txtPassword.Text;
            }
            set
            {
                txtPassword.Text = value;
                txtReenter.Text = value;
            }
        }        

        /// <summary>
        /// Gets or sets the encryption algorithm.
        /// </summary>
        public EncryptionAlgorithm EncryptionAlgorithm
        {
            get 
            {
                if (txtPassword.Text.Length == 0)
                    return EncryptionAlgorithm.None;

                switch (cboEncryption.SelectedIndex)
                {
                    case 0:
                        return EncryptionAlgorithm.PkzipClassic;

                    case 1:
                        return EncryptionAlgorithm.Aes128;

                    case 2:
                        return EncryptionAlgorithm.Aes192;

                    default:
                        return EncryptionAlgorithm.Aes256;
                }
            }
            set
            {
                switch (value)
                {
                    case EncryptionAlgorithm.None:
                    case EncryptionAlgorithm.PkzipClassic:
                        cboEncryption.SelectedIndex = 0; break;

                    case EncryptionAlgorithm.Aes128:
                        cboEncryption.SelectedIndex = 1; break;

                    case EncryptionAlgorithm.Aes192:
                        cboEncryption.SelectedIndex = 2; break;

                    case EncryptionAlgorithm.Aes256:
                        cboEncryption.SelectedIndex = 3; break;
                }                
            }
        }

        /// <summary>
        /// Handles the OK button's Click event.
        /// </summary>
        /// <param name="sender">The button object.</param>
        /// <param name="e">The event arguments.</param>
        private void btnOK_Click(object sender, EventArgs e)
        {
            // Check passwords.
            if (txtReenter.Text != txtPassword.Text)
            {
                MessageBox.Show("Reentered password does not match with password!", "ArchiveManager", MessageBoxButtons.OK, MessageBoxIcon.Information);
                return;
            }

            DialogResult = DialogResult.OK;
        }
    }
}

File: Zipping and Unzipping files in .NET application/[VB]-Zipping and Unzipping files in .NET application/VB/Password.vb
Imports ComponentPro.Compression

Namespace ArchiveManager
	Partial Public Class Password
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		''' <summary>
		''' Gets or sets the archive's password.
		''' </summary>
		Public Property Pass() As String
			Get
				Return txtPassword.Text
			End Get
			Set(ByVal value As String)
				txtPassword.Text = value
				txtReenter.Text = value
			End Set
		End Property

		''' <summary>
		''' Gets or sets the encryption algorithm.
		''' </summary>
		Public Property EncryptionAlgorithm() As EncryptionAlgorithm
			Get
				If txtPassword.Text.Length = 0 Then
					Return EncryptionAlgorithm.None
				End If

				Select Case cboEncryption.SelectedIndex
					Case 0
						Return EncryptionAlgorithm.PkzipClassic

					Case 1
						Return EncryptionAlgorithm.Aes128

					Case 2
						Return EncryptionAlgorithm.Aes192

					Case Else
						Return EncryptionAlgorithm.Aes256
				End Select
			End Get
			Set(ByVal value As EncryptionAlgorithm)
				Select Case value
					Case EncryptionAlgorithm.None, EncryptionAlgorithm.PkzipClassic
						cboEncryption.SelectedIndex = 0

					Case EncryptionAlgorithm.Aes128
						cboEncryption.SelectedIndex = 1

					Case EncryptionAlgorithm.Aes192
						cboEncryption.SelectedIndex = 2

					Case EncryptionAlgorithm.Aes256
						cboEncryption.SelectedIndex = 3
				End Select
			End Set
		End Property

		''' <summary>
		''' Handles the OK button's Click event.
		''' </summary>
		''' <param name="sender">The button object.</param>
		''' <param name="e">The event arguments.</param>
		Private Sub btnOK_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnOK.Click
			' Check passwords.
			If txtReenter.Text <> txtPassword.Text Then
				MessageBox.Show("Reentered password does not match with password!", "ArchiveManager", MessageBoxButtons.OK, MessageBoxIcon.Information)
				Return
			End If

			DialogResult = System.Windows.Forms.DialogResult.OK
		End Sub
	End Class
End Namespace